File: 18-08-2025_Find_H_Index.java
class Solution {
    private boolean check(int[] arr, int H) {
        int count = 0;
        for (int x : arr) {
            if (x >= H) count++;
        }
        return count >= H;
    }

    public int hIndex(int[] citations) {
        int low = 1, high = 0;
        for (int c : citations) {
            high = Math.max(high, c);
        }

        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (check(citations, mid)) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }

        return high;
    }
}

File: 18-08-2025_Find_H_Index.py
class Solution:
    def hIndex(self, citations: list[int]) -> int:
        def check(arr, H):
            count = 0
            for x in arr:
                if x >= H:
                    count += 1
            return count >= H

        low, high = 1, max(citations) if citations else 0

        while low <= high:
            mid = low + (high - low) // 2
            if check(citations, mid):
                low = mid + 1
            else:
                high = mid - 1

        return high
